gotham/src/router/builder/single.rs

```diff
@@ -157,6 +175,31 @@ pub trait DefineSingleRoute {
         Self: Sized,
         H: (FnOnce(State) -> Fut) + RefUnwindSafe + Copy + Send + Sync + 'static,
         Fut: Future<Output = HandlerResult> + Send + 'static;
+
+    /// Similar to `to_async`, but passes in State as a reference
+    fn to_async_borrowing<F, R: IntoResponse + 'static>(self, handler: F)
+    where
+        Self: Sized,
+        for<'a> F: AsyncHandlerFn<'a, Res = R> + RefUnwindSafe + Clone + Send + Sync + 'static,
+    {
+        self.to_new_handler(move || {
+            let handler = handler.clone();
+            Ok(move |mut state: State| {
+                async move {
+                    let fut = handler.call(&mut state);
+                    let result = fut.await;
+                    match result {
+                        Ok(data) => {
+                            let response = data.into_response(&state);
+                            Ok((state, response))
+                        }
+                        Err(err) => Err((state, err)),
+                    }
+                }
+                .boxed()
```

Comment:
is it possible to move this up into the AsyncHandlerFn as a `call_wrapped(State) -> Box<dyn Future<Output = Result<(State, _), (State, _)>>`? At the moment, there are a lot of complex moving parts that are spread all over the file.

Reply:
It may be possible to move this into a free function or another trait, but it explicitly cannot be moved into `AsyncHandlerFn` trait itself because it uses HRTB on that.